docs/source/commands/contracts.mdx

```diff
@@ -0,0 +1,183 @@
+---
+title: Rover contract commands
+description: For use with Apollo Contract variants
+---
+
+A **contract** is a variant that is the result of filtering out schema elements from a source variant.
+
+Rover commands that interact with [contracts](https://www.apollographql.com/docs/graphos/delivery/contracts/) begin with `rover contract`.
+
+## Publishing a contract to Apollo Studio
+
+### `contract publish`
+
+> This command requires [authenticating Rover with Apollo Studio](../configuring/#authenticating-with-apollo-studio).
+
+You can use Rover to publish a new contract or publish configuration changes to an existing contract.
+
+Run the `contract publish` command, like so:
+
+```bash
+rover contract publish my-graph@my-contract-variant \
+  --source-variant my-source-variant \
+  --include foo --include bar \
+  --exclude baz \
+  --hide-unreachable-types
+```
+
+The argument `my-graph@my-contract-variant` in the example above is a [graph ref](../conventions/#graph-refs) that specifies the ID of the Studio graph you're publishing to, along with which [variant](/graphos/graphs/overview/#variants) you're publishing to.
+
+If this variant already exists in the graph registry, its configuration is updated; otherwise, a new contract variant is created.
+
+Options include:
+
+<table class="field-table">
+<thead>
+<tr>
+<th>Name</th>
+<th>Description</th>
+</tr>
+</thead>
+<tbody>
+<tr class="required">
+<td>
+
+###### `--source-variant`
+
+</td>
+<td>
+
+The name of the [source variant](https://www.apollographql.com/docs/graphos/delivery/contracts/#2-fed1-enable-variant-support-for-tag) whose supergraph schema is used for filtering.
+
+This variant must be on the same graph as the contract variant and cannot be changed after contract creation. It must also be a federated variant with subgraphs, and if using Fed 1, support for the `@tag` directive must be explicitly enabled in Studio.
```

Comment:
I think it makes sense to split this up. Maybe even move the federation 1 bit into a sub-note in small italics.

```suggestion
This variant must be on the same graph as the contract variant and cannot be changed after contract creation. It must also be a federated variant with subgraphs. If your graph is using Federation 1, support for the `@tag` directive must be explicitly enabled in Studio.
```